iothub/device/samples/PnpDeviceSamples/TemperatureController/TemperatureControllerSample.cs

```diff
@@ -0,0 +1,316 @@
+
+        public TemperatureControllerSample(DeviceClient deviceClient, ILogger logger)
+        {
+            _deviceClient = deviceClient ?? throw new ArgumentNullException($"{nameof(deviceClient)} cannot be null.");
+            _logger = logger ?? LoggerFactory.Create(builer => builer.AddConsole()).CreateLogger<TemperatureControllerSample>();
+        }
+
+        public async Task PerformOperationsAsync(CancellationToken cancellationToken)
+        {
+            // This sample follows the following workflow:
+            // -> Set handler to receive "reboot" command - root interface.
+            // -> Set handler to receive "getMaxMinReport" command - on "Thermostat" components.
+            // -> Set handler to receive "targetTemperature" property updates from service - on "Thermostat" components.
+            // -> Send initial device info - "workingSet" over telemetry, "serialNumber" over reported property update - root interface.
+            // -> Periodically send "temperature" over telemetry - on "Thermostat" components.
+            // -> Send "maxTempSinceLastReboot" over property update, when a new max temperature is set - on "Thermostat" components.
+
+            _logger.LogDebug($"Set handler for \"reboot\" command.");
+            await _deviceClient.SetMethodHandlerAsync("reboot", HandleRebootCommandAsync, _deviceClient, cancellationToken);
+
+            _logger.LogDebug($"Set handler for \"getMaxMinReport\" command.");
+            await _deviceClient.SetMethodHandlerAsync("thermostat1*getMaxMinReport", HandleMaxMinReportCommandAsync, Thermostat1, cancellationToken);
+            await _deviceClient.SetMethodHandlerAsync("thermostat2*getMaxMinReport", HandleMaxMinReportCommandAsync, Thermostat2, cancellationToken);
+
+            _logger.LogDebug($"Set handler to receive \"targetTemperature\" updates.");
+            await _deviceClient.SetDesiredPropertyUpdateCallbackAsync(SetDesiredPropertyUpdateCallbackAsync, null, cancellationToken);
+            _desiredPropertyUpdateCallbacks.Add(Thermostat1, TargetTemperatureUpdateCallbackAsync);
+            _desiredPropertyUpdateCallbacks.Add(Thermostat2, TargetTemperatureUpdateCallbackAsync);
+
+            await SendDeviceMemoryAsync(cancellationToken);
+            await SendDeviceSerialNumberAsync(cancellationToken);
+
+            bool temperatureReset = true;
+            _maxTemp[Thermostat1] = 0d;
+            _maxTemp[Thermostat2] = 0d;
+
+            await Task.Run(async () =>
+            {
+                while (!cancellationToken.IsCancellationRequested)
+                {
+                    if (temperatureReset)
+                    {
+                        // Generate a random value between 5.0°C and 45.0°C for the current temperature reading for each "Thermostat" component.
+                        _temperature[Thermostat1] = Math.Round(_random.NextDouble() * 40.0 + 5.0, 1);
```

Comment:
DRY

Reply:
We get the next random double, and then round it to 1 decimal place. Is there a better way to do this?